backend/src/data/migrations/1616023319930-components-entities.ts

```diff
@@ -0,0 +1,44 @@
+import {MigrationInterface, QueryRunner} from "typeorm";
+
+export class componentsEntities1616023319930 implements MigrationInterface {
+    name = 'componentsEntities1616023319930'
+
+    public async up(queryRunner: QueryRunner): Promise<void> {
+        await queryRunner.query(`CREATE TABLE "table" ("id" uuid NOT NULL DEFAULT uuid_generate_v4(), "createdAt" TIMESTAMP NOT NULL DEFAULT now(), "updatedAt" TIMESTAMP NOT NULL DEFAULT now(), "componentId" uuid NOT NULL, "queryId" uuid NOT NULL, CONSTRAINT "REL_ca7d1088a45877e520ede2042e" UNIQUE ("componentId"), CONSTRAINT "PK_28914b55c485fc2d7a101b1b2a4" PRIMARY KEY ("id"))`);
+        await queryRunner.query(`CREATE TABLE "button" ("id" uuid NOT NULL DEFAULT uuid_generate_v4(), "createdAt" TIMESTAMP NOT NULL DEFAULT now(), "updatedAt" TIMESTAMP NOT NULL DEFAULT now(), "componentId" uuid NOT NULL, "queryId" uuid NOT NULL, "label" character varying NOT NULL, CONSTRAINT "REL_fb96e8d5571a89dbce61dcb52f" UNIQUE ("componentId"), CONSTRAINT "PK_a4df4e4f7a5882bc94442d3f209" PRIMARY KEY ("id"))`);
+        await queryRunner.query(`CREATE TABLE "input" ("id" uuid NOT NULL DEFAULT uuid_generate_v4(), "createdAt" TIMESTAMP NOT NULL DEFAULT now(), "updatedAt" TIMESTAMP NOT NULL DEFAULT now(), "componentId" uuid NOT NULL, "label" character varying NOT NULL, "placeholder" character varying NOT NULL, CONSTRAINT "REL_c0eab412e9d37c87d157601c7b" UNIQUE ("componentId"), CONSTRAINT "PK_a1deaa2fcdc821329884ad43931" PRIMARY KEY ("id"))`);
+        await queryRunner.query(`CREATE TABLE "text" ("id" uuid NOT NULL DEFAULT uuid_generate_v4(), "createdAt" TIMESTAMP NOT NULL DEFAULT now(), "updatedAt" TIMESTAMP NOT NULL DEFAULT now(), "componentId" uuid NOT NULL, "value" character varying NOT NULL, CONSTRAINT "REL_f9e02d9a23e6b62bfcf712e7cd" UNIQUE ("componentId"), CONSTRAINT "PK_ef734161ea7c326fedf699309f9" PRIMARY KEY ("id"))`);
+        await queryRunner.query(`ALTER TABLE "user" ALTER COLUMN "firstName" SET NOT NULL`);
+        await queryRunner.query(`COMMENT ON COLUMN "user"."firstName" IS NULL`);
+        await queryRunner.query(`ALTER TABLE "user" ALTER COLUMN "lastName" SET NOT NULL`);
+        await queryRunner.query(`COMMENT ON COLUMN "user"."lastName" IS NULL`);
+        await queryRunner.query(`ALTER TABLE "user" ALTER COLUMN "password" SET NOT NULL`);
+        await queryRunner.query(`COMMENT ON COLUMN "user"."password" IS NULL`);
```

Comment:
We shouldn't have these lines in the migration, these columns should remain nullable. Could you please check the User entity and make these columns nullable there? And then regenerate this migration.
Everything else looks good!